jobs/jobscheduler.py

```diff
@@ -70,8 +71,23 @@ def runJob(job):
         job["exit_code"] = 0
     else:
         cmd = job["command"]
-        if ("gpu" in job) and (job["gpu"] >= 0):
-            cmd = cmd.replace("HIP_VISIBLE_DEVICES=0", "HIP_VISIBLE_DEVICES=%s" % job["gpu"])
+        if job.get("gpu", -1) >= 0:
+            maker, idx = job["gpu"].split(":", 1)
+            # Remove assignment of GPUs if it exists
+            cmd = re.sub(r"HIP_VISIBLE_DEVICES=\S+", "", cmd)
+            cmd = re.sub(r"CUDA_VISIBLE_DEVICES=\S+", "", cmd)
+
+            if maker == "cuda":
+                cmd = cmd.replace(
+                    "cmsDriver.py",
+                    "CUDA_VISIBLE_DEVICES=%s HIP_VISIBLE_DEVICES= cmsDriver.py" % idx,
+                )
+            elif maker == "rocm":
+                cmd = cmd.replace(
+                    "cmsDriver.py",
+                    "HIP_VISIBLE_DEVICES=%s CUDA_VISIBLE_DEVICES= cmsDriver.py" % idx,
+                )
```

Comment:
@fwyzard , should we support `CUDA_VISIBLE_DEVICES=N HIP_VISIBLE_DEVICES=N` or will we only request one type of gpu? I mean in future can runTheMatrix request to run a workflow step to use both amd and nvidia gpus?